components/sup/src/command/start.rs

```diff
@@ -50,109 +50,41 @@
-                    let depot_client = try!(Client::new(&spec.depot_url, PRODUCT, VERSION, None));
-                    let latest_pkg_data = try!(depot_client.show_package(&spec.ident));
-                    let latest_ident: PackageIdent = latest_pkg_data.get_ident().clone().into();
-                    if &latest_ident > package.ident() {
-                        outputln!("Downloading latest version from Depot: {}", latest_ident);
-                        let new_pkg_data = try!(install::start(&mut ui,
-                                                               &spec.depot_url,
-                                                               &latest_ident.to_string(),
-                                                               PRODUCT,
-                                                               VERSION,
-                                                               Path::new(&*FS_ROOT_PATH),
-                                                               &cache_artifact_path(None),
-                                                               false));
-                        package = try!(PackageInstall::load(&new_pkg_data, Some(&*FS_ROOT_PATH)));
-                    } else {
-                        outputln!("Already running latest.");
-                    };
-                }
-            }
-            start_package(package, cfg, spec)
-        }
-        Err(_) => {
-            outputln!("{} is not installed",
-                      Yellow.bold().paint(spec.ident.to_string()));
-            let new_pkg_data = match local_artifact {
-                Some(artifact) => {
-                    try!(install::start(&mut ui,
-                                        &spec.depot_url,
-                                        &artifact,
-                                        PRODUCT,
-                                        VERSION,
-                                        Path::new(&*FS_ROOT_PATH),
-                                        &cache_artifact_path(None),
-                                        false))
-                }
-                None => {
-                    outputln!("Searching for {} in remote {}",
-                              Yellow.bold().paint(spec.ident.to_string()),
-                              &spec.depot_url);
-                    try!(install::start(&mut ui,
-                                        &spec.depot_url,
-                                        &spec.ident.to_string(),
-                                        PRODUCT,
-                                        VERSION,
-                                        Path::new(&*FS_ROOT_PATH),
-                                        &cache_artifact_path(None),
-                                        false))
-                }
-            };
-            let package = try!(PackageInstall::load(&new_pkg_data, Some(&*FS_ROOT_PATH)));
-            start_package(package, cfg, spec)
-        }
+    if let Some(artifact) = local_artifact {
```

Comment:
Yeah, I agree with that, especially the issue of adding on rule exceptions that explode with *bigly* over time. In trying to avoid that I did 2 things:

* Leave any notion of a local artifact out of the `ServiceSpec`--the artifact's fully qualified ident is used for the `ServiceSpec`'s ident and that's it
* Treat the local artifact case quickly by naiively installing it to disk and moving on, in effect "boiling off" the fact that a local artifact was asked for which is why it never leaves the command starting stage and enters the library internals (I've been thinking of it similar to an environment variable--get it gone as soon as possible before you call too deep)

With these 2 tricks I don't see a way we'd enable an end user to provide a config file (mapping to a `ServiceSpec`) that has an artifact file in it or allow any update strategies to work (you asked to run that specific ident and nothing else).